src/org/opensolaris/opengrok/configuration/Configuration.java

```diff
@@ -421,15 +458,35 @@ public int getCachePages() {
         return cachePages;
     }
 
-    public void setCachePages(int cachePages) {
+    /**
+     * Set the cache pages to a new value
+     *
+     * @param cachePages the new value
+     * @throws IllegalArgumentException when the cachePages is negative
+     */
+    public void setCachePages(int cachePages) throws IllegalArgumentException {
+        if (cachePages < 0) {
+            throw new IllegalArgumentException(
+                    String.format("Invalid value for cachePages \"%d\"", cachePages));
+        }
         this.cachePages = cachePages;
     }
 
     public int getHitsPerPage() {
         return hitsPerPage;
     }
 
-    public void setHitsPerPage(int hitsPerPage) {
+    /**
+     * Set the hits per page to a new value
+     *
+     * @param hitsPerPage the new value
+     * @throws IllegalArgumentException when the hitsPerPage is negative
+     */
+    public void setHitsPerPage(int hitsPerPage) throws IllegalArgumentException {
+        if (hitsPerPage < 0) {
+            throw new IllegalArgumentException(
+                    String.format("Invalid value for histPerPage \"%d\"", hitsPerPage));
```

Comment:
this and all of the above cases that check for negative number could be de-duped. Share the error string, consider new special exception type, etc.